src/ComputationContainer/Logging/Logger.hpp

```diff
@@ -1,92 +1,139 @@
 #pragma once
+#include <iostream>
+#include <memory>
+#include <optional>
+#include <string_view>
+#include <utility>
+
 #include <boost/exception/all.hpp>
 #include <boost/format.hpp>
 #include <boost/stacktrace.hpp>
-#include <ctime>
-#include <iomanip>
-#include <iostream>
-#include <sstream>
-#include <variant>
+#include <boost/stacktrace/stacktrace_fwd.hpp>
+
+#include "spdlog/sinks/dist_sink.h"
+#include "spdlog/sinks/stdout_color_sinks.h"
+
+#include "LogHeader/Logger.hpp"
+
+#define QMPC_LOG_LOCATION() \
+    spdlog::source_loc { __FILE__, __LINE__, SPDLOG_FUNCTION }
+#define QMPC_LOGGER_CALL(func, ...) func(QMPC_LOG_LOCATION(), __VA_ARGS__)
+#define QMPC_LOG_INFO(...) QMPC_LOGGER_CALL(qmpc::Log::Info, __VA_ARGS__)
+#define QMPC_LOG_DEBUG(...) QMPC_LOGGER_CALL(qmpc::Log::Debug, __VA_ARGS__)
+#define QMPC_LOG_ERROR(...) QMPC_LOGGER_CALL(qmpc::Log::Error, __VA_ARGS__)
+#define QMPC_LOG_WARN(...) QMPC_LOGGER_CALL(qmpc::Log::Warn, __VA_ARGS__)
 
 namespace qmpc
 {
-// TODO: std::coutをspdlogに差し替え
-// TODO: クラスにする必要がないが念のためクラス構造にしておく
 class Log
 {
-    enum class LogLevel
-    {
-        Info,
-        Debug,
-        Error
-    };
-    Log::LogLevel level;
-    static inline std::unordered_map<qmpc::Log::LogLevel, std::string> logLevelStr = {
-        {Log::LogLevel::Info, "INFO"},
-        {Log::LogLevel::Debug, "DEBUG"},
-        {Log::LogLevel::Error, "ERROR"}};
-
 private:
-    std::string getTime() const
+    static constexpr auto LOG_FORMAT =
+        std::string_view("%Y-%m-%d %T %z | %^%-5l%$ | %g:%!:%# - %v");
+    std::shared_ptr<spdlog::sinks::dist_sink_mt> sinks;
+    std::shared_ptr<spdlog::logger> logger;
+    Log()
+        : sinks(std::make_shared<spdlog::sinks::dist_sink_mt>())
+        , logger(std::make_shared<spdlog::logger>("QMPC Logger", sinks))
```

Comment:
```suggestion
        : sinks(std::make_unique<spdlog::sinks::dist_sink_mt>())
        , logger(std::make_unique<spdlog::logger>("QMPC Logger", sinks))
```

Reply:
Changed only logger to unique ptr.
See https://github.com/acompany-develop/QuickMPC/commit/3263e64a2014d77c08d4adb75e372e435f6c8fd7 more details.